frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -0,0 +1,239 @@
+import { useState, useEffect } from "react";
+import { t, jt } from "ttag";
+import _ from "underscore";
+
+import ErrorBoundary from "metabase/ErrorBoundary";
+import Alert from "metabase/core/components/Alert";
+import Link from "metabase/core/components/Link";
+import {
+  Form,
+  FormCheckbox,
+  FormProvider,
+  FormSubmitButton,
+} from "metabase/forms";
+import { useToggle } from "metabase/hooks/use-toggle";
+import { capitalize } from "metabase/lib/formatting";
+import {
+  Button,
+  Center,
+  Icon,
+  Loader,
+  Text,
+  Stack,
+  Box,
+  Modal,
+  Flex,
+} from "metabase/ui";
+
+import type { ErrorPayload } from "./types";
+import { useErrorInfo } from "./use-error-info";
+import { downloadObjectAsJson, hasQueryData } from "./utils";
+
+interface ErrorDiagnosticModalProps {
+  errorInfo: ErrorPayload;
+  onClose: () => void;
+}
+
+type PayloadSelection = Partial<Record<keyof ErrorPayload, boolean>>;
+
+export const ErrorDiagnosticModal = ({
+  errorInfo,
+  onClose,
+}: ErrorDiagnosticModalProps) => {
+  const handleSubmit = (values: PayloadSelection) => {
+    const selectedKeys = Object.keys(values).filter(
+      key => values[key as keyof PayloadSelection],
+    );
+    const selectedInfo: ErrorPayload = _.pick(errorInfo, ...selectedKeys);
+
+    downloadObjectAsJson(
+      selectedInfo,
+      `metabase-diagnostic-info-${new Date().toISOString()}`,
+    );
+    onClose();
+  };
+
+  const canIncludeQueryData = hasQueryData(errorInfo?.entityName);
+
+  const hiddenValues = {
+    url: true,
+    entityName: true,
+  };
+
+  return (
+    <Modal
+      opened
+      onClose={onClose}
+      title={t`Download diagnostic information`}
+      padding="xl"
+      size="lg"
+    >
+      <FormProvider
+        initialValues={{
+          ...hiddenValues,
+          queryResults: false,
+          entityInfo: true,
+          frontendErrors: true,
+          backendErrors: true,
+          userLogs: true,
+          logs: true,
+          instanceInfo: true,
+        }}
+        onSubmit={handleSubmit}
+      >
+        <Form>
+          <Text>
+            {t`Select the info you want to include in the diagnostic JSON file.`}
```

Comment:
perhaps: "Select the information you want to include:"

Reply:
see [slack thread re. copy](https://metaboat.slack.com/archives/C064EB1UE5P/p1709751447677899?thread_ts=1709749714.608869&cid=C064EB1UE5P)

Reply:
I still like the shorter, simpler phrase fwiw

When I imagine a real person saying these words to me, I think it would be strange of them to allude to "the JSON file" as though I'm already supposed to know what that is. But perhaps I'm missing some context